### PR TITLE
add easyblock for ADF (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/easyblocks/a/adf.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    def build_step(self):
+        """No separate custom build procedure for ADF, see install step."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for ADF."""
+
+        # this shouldn't be needed, is init.sh shipped in the wrong place?!
+        shutil.copy2(os.path.join('Install', 'init.sh'), os.path.join('bin', 'init.sh'))
```
Comment (boegel) [inline easybuild/easyblocks/a/adf.py]: unclear why this is needed

File: easybuild/easyblocks/a/adf.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        # strip off 'adf<version>' part to avoid having everything in a subdirectory
+        self.cfg['unpack_options'] = "--strip-components=1"
+        super(EB_ADF, self).extract_step()
+
+    def configure_step(self):
+        """Custom configuration procedure for ADF."""
+
+        env.setvar('ADFHOME', self.installdir)
+        env.setvar('ADFBIN', os.path.join(self.installdir, 'bin'))
+        env.setvar('ADFRESOURCES', os.path.join(self.installdir, 'atomicdata'))
```
Comment (boegel) [inline easybuild/easyblocks/a/adf.py]: this seems to be wrong `atomicdata` isn't there?

File: easybuild/easyblocks/a/adf.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        shutil.copy2(os.path.join('Install', 'init.sh'), os.path.join('bin', 'init.sh'))
+
+        cmd = "./bin/foray -j %d" % self.cfg['parallel']
+        run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def sanity_check_step(self):
+        """Custom sanity check for ADF."""
+        custom_paths = {
+            'files': ['bin/adf'],
+            'dirs': ['atomicdata', 'examples'],
```
Comment (boegel) [inline easybuild/easyblocks/a/adf.py]: currently fails, `atomicdata` and `example` are not there like they are in the binary version...

File: easybuild/easyblocks/a/adf.py (hunk 2 above)
Comment (boegel) [inline easybuild/easyblocks/a/adf.py]: ah, seems like you need to unpack both the sources and binary version, and then recompile

http://www.scm.com/Doc/Doc2014/Install/Installation/page5.html

that explains why some stuff is missing

File: easybuild/easyblocks/a/adf.py (hunk 3 above)
Comment (boegel) [inline easybuild/easyblocks/a/adf.py]: fixed in easyconfig, both sources and binary version are available to recompile from source >_<

**Comment by boegel** [comment]
@wpoely86: please review?

File: easybuild/easyblocks/a/adf.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing ADF, implemented as an easyblock
```
Comment (wpoely86) [inline easybuild/easyblocks/a/adf.py]: `@author ` tag

**Comment by wpoely86** [comment]
lgtm

**Comment by boegel** [comment]
Thanks for the review @wpoely86!

**Comment by boegel** [comment]
Tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/2451, works as intended, so good to go.

Thanks for the review @wpoely86!

**Pull request merged**